cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -208,6 +209,7 @@ static PyObject *log_msg(PyObject *self, PyObject *args)
 // Remaining arguments are keyword arguments to be passed to the callback
 static PyObject *register_readonly_callback(PyObject *self, PyObject *args)
 {
+    (void)self;
```

Comment:
I think we might want a cleaner solution to this problem, perhaps
```
#define COCOTB_UNUSED(x) __attribute__((unused)) x
```
which you can then use inside the function signature. Worth looking at how other projects solve this problem cleanly.

Reply:
If all the sources were C++ you could just leave off the parameter name. But since they are not, we could use the void cast or the unused attribute. But there [doesn't seem to be any difference](https://godbolt.org/z/y8TauJ), and the attribute yields longer function specs that have to be duplicated at declaration and definition, and is an extension.

Reply:
There's obviously no difference in compiler output, my argument stems from readability. To a newcomer, `UNUSED` is easier to understand than `(void)`, especially if you out an explanatory comment above the macro definition.

> function specs that have to be duplicated at declaration and definition

Are you sure this is true?

> and is an extension.

Hence the suggestion to wrap it in a macro. Searching online finds many much better macros, the one I suggested was just an example.